pkg/controllers/dashboardapi/settings/settings_test.go

```diff
@@ -257,9 +250,52 @@ func populateTestCases() []*testCase {
 		}
 		test.existingSetting.Name = settingName
 	}
+
 	return testCases
 }
 
-func stringPtr(val string) *string {
-	return &val
+func TestMarkSettingAsUnknownRetry(t *testing.T) {
+	store := map[string]v3.Setting{
+		"unknown": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "unknown",
+			},
+			Value:   "unknown",
+			Default: "unknown",
+		},
+	}
+
+	client := fake.NewMockNonNamespacedControllerInterface[*v3.Setting, *v3.SettingList](gomock.NewController(t))
+
+	client.EXPECT().Get(gomock.Any(), gomock.Any()).DoAndReturn(func(name string, options metav1.GetOptions) (*v3.Setting, error) {
+		val, ok := store[name]
+		if !ok {
+			return nil, apierrors.NewNotFound(schema.GroupResource{}, name)
+		}
+
+		return &val, nil
+	}).Times(1)
+
+	var updateRun int
+	client.EXPECT().Update(gomock.Any()).DoAndReturn(func(s *v3.Setting) (*v3.Setting, error) {
+		defer func() { updateRun++ }()
+
+		if updateRun == 0 { // Fail the the first update to force retry.
+			return nil, apierrors.NewConflict(schema.GroupResource{}, s.Name, fmt.Errorf("some error"))
```

Comment:
Nit: It's a good practice to provide as realistic returns as possible. In this case, that would mean returning the GroupResource for the setting CRD (`management.cattle.io` and `settings`).

Reply:
Agreed.